src/Sarif/RuleKindSet.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.Serialization;
+
+using Newtonsoft.Json;
+
+namespace Microsoft.CodeAnalysis.Sarif
+{
+    [Serializable]
+    [JsonConverter(typeof(TypedPropertiesDictionaryConverter))]
+    public class RuleKindSet : HashSet<RuleKind>
```

Comment:
>RuleKindSet [](http://example.com/codeflow?start=17&length=11)

Why couldn't we use a HashSet?

Reply:
this is following the same pattern as other fields we have, the ResultKind.
The commanline nuget package the options have some limitation so there is some conversions needed